vision_ocv/benchmark/modules/src/BM_ocv_vision_factory_test.cpp

```diff
@@ -0,0 +1,168 @@
+        }
+
+        if (useEventLoop) {
+            _eventLoopProvider.reset(
+                new kpsr::high_performance::EventLoopMiddlewareProvider<16>(nullptr, "img"));
+            _eventLoopProvider->start();
+
+            subscriber = _eventLoopProvider->getSubscriber<kpsr::vision_ocv::ImageData>(
+                "imageEvent");
+            publisher = _eventLoopProvider->getPublisher<kpsr::vision_ocv::ImageData>("imageEvent",
+                                                                                      poolSize,
+                                                                                      _buildF,
+                                                                                      _cloneF);
+        } else {
+            _eventEmitterProvider.reset(
+                new kpsr::EventEmitterMiddlewareProvider<kpsr::vision_ocv::ImageData>(nullptr,
+                                                                                      "img",
+                                                                                      poolSize,
+                                                                                      _buildF,
+                                                                                      _cloneF));
+
+            subscriber = _eventEmitterProvider->getSubscriber();
+            publisher = _eventEmitterProvider->getPublisher();
+        }
+    }
+
+    void TearDown(const ::benchmark::State &state)
+    {
+        subscriber->removeListener("receiveIm");
+        bool useEventLoop = state.range(2);
+        if (useEventLoop) {
+            _eventLoopProvider->stop();
+        }
+    }
+
+    void setListener(benchmark::State &state)
+    {
+        subscriber->registerListener("receiveIm", [&state](const kpsr::vision_ocv::ImageData &im) {
+            auto now = std::chrono::duration_cast<std::chrono::milliseconds>(
+                           std::chrono::system_clock::now().time_since_epoch())
+                           .count();
+            state.SetIterationTime(now - im.timestamp);
+        });
+    }
+    int width, height;
+    kpsr::vision_ocv::ImageDataFactory imageDataFactory;
+    kpsr::Subscriber<kpsr::vision_ocv::ImageData> *subscriber;
+    kpsr::Publisher<kpsr::vision_ocv::ImageData> *publisher;
+
+private:
+    std::function<void(const kpsr::vision_ocv::ImageData &, kpsr::vision_ocv::ImageData &)> _cloneF;
+    std::function<void(kpsr::vision_ocv::ImageData &)> _buildF;
+    std::unique_ptr<kpsr::EventEmitterMiddlewareProvider<kpsr::vision_ocv::ImageData>>
+        _eventEmitterProvider;
+    std::unique_ptr<kpsr::high_performance::EventLoopMiddlewareProvider<16>> _eventLoopProvider;
+};
+
+BENCHMARK_DEFINE_F(FactoryPerformanceTest, SimpleTest)(benchmark::State &state)
+{
+    bool useEventLoop = state.range(2);
+    setListener(state);
+
+    auto &publisherStats = publisher->_publicationStats;
+    auto listenerStats = subscriber->getSubscriptionStats("receiveIm");
```

Comment:
The code in kpsr-core should be fixed, but as of now, listenerStats could be a nullptr in some cases if the container is null :https://github.com/klepsydra-technologies/kpsr-core/blob/main/core/modules/core/include/klepsydra/core/unsafe_multi_listener_event_emitter.h#L72 . 

So for the time being, add simple atomic counters in the benchmarking for loop, and the listener. this will prevent segfaults. Log those counters rather than the publisher Stats. 

Finally, add a todo here to switch to using publisher/subscriber stats when kpsr-core is fixed.